### PR TITLE
Move theme files to cache dir

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR will move `_themes` from `site-package/napari/resources/icons/_themes` to `cache-dir/napari/_themes`.

This should solve the problem of preventing napari from running after accidentally installing napari without `-e` option and then trying to use it in editable mode.

Minor fixes in formating

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/resources/_icons.py [hunk 1]
```diff
@@ -138,7 +139,7 @@ def write_colorized_svgs(
     colors: Iterable[Union[str, Tuple[str, str]]],
     opacities: Iterable[float] = (1.0,),
     theme_override: Optional[Dict[str, str]] = None,
-) -> Iterator[Tuple[str, str]]:
+):
```
Comment (jni) [inline napari/resources/_icons.py:142]: Is this change intended @Czaki?

File: napari/resources/_icons.py [hunk 2]
```diff
@@ -154,7 +155,7 @@ def write_colorized_svgs(
 
 
 def _theme_path(theme_name: str) -> Path:
-    return ICON_PATH / '_themes' / theme_name
+    return Path(user_cache_dir()) / '_themes' / theme_name
```
Comment (jni) [inline napari/resources/_icons.py:158]: Does this mean now that multiple napari versions (e.g. from different conda environments) will share the same theme directory? That might be a problem? Should there perhaps be a hash of the version or of the executable path there?

File: napari/resources/_icons.py (hunk 1 above)
Comment (Czaki) [reply napari/resources/_icons.py:142]: I made this change as this function returns nothing.

File: napari/resources/_icons.py (hunk 2 above)
Comment (Czaki) [reply napari/resources/_icons.py:158]: This is the same problem as with settings.

Comment (Czaki) [reply napari/resources/_icons.py:158]: I change to cache dir be separate per version

Comment (jni) [reply napari/resources/_icons.py:158]: Yeah but settings has explicit safeguards (?) for incorrect or deprecated keys, where I don't think the theme does? Anyway thanks for the update! 🙏

**Review by jni — APPROVED** [review]

**Comment by andy-sweet** [comment]
I just want to clarify the issue this fixes before cherry-picking for v0.4.17. Currently, if the user follows these steps

1. Create Python environment with Python 3.10
2. Install napari (i.e. at this time v0.4.16 is stable
3. Install napari as editable (likely after cloning the repo and working from the `main` branch)

then they'll get a confusing error that can be fixed by deleting some files in their conda directory. I reproduced that issue and verified the workaround too.

If the user has already installed v0.4.16 in their environment, and has a repo that contains the changes in this PR, they'll still run into the same issue.

But if the first napari they install in their environment is v0.4.17 and that release includes the changes in this PR, then they won't get the error. Which is why we should consider cherry-picking this PR.

Is my understanding correct? If so, I think we should cherry-pick this into v0.4.17.

**Comment by Czaki** [comment]
Yes

**Comment by andy-sweet** [comment]
OK, I'm going to merge this after 24 hours if no one else. And will then cherry pick it to v0.4.17x.
